### DISCUSSION TITLE
Tutorial problems with Constrain Workplane Normal and Create Regular Polygon

### DISCUSSION THREAD

**Discussion by osresearch** [dune3d/dune3d#75] [Q&A] [answered]
I had trouble with the `Constrain Workplane Normal` [step in the example part tutorial](https://docs.dune3d.org/en/latest/simple-part.html#create-workplane).  I created the workplane on the midpoint of the construction diagonal, then click on the bottom edge of the face (the one on the XY plane).  The tool tip updates to "select other vector" and when I click on one of the verticals from the extrusion it says "please click a line from a previous group". By looking at the tree in the side bar of the tutorial, I realized that there was a missing step for "Create a new sketch". Adding a new sketch allowed me to align the workplane to those edges from the earlier groups.

Now that my workplane is aligned, I can't `Create Regular Polygon`.  The controls are grayed out, except for create workplane one, and hitting space for the context menu only as draw line in 3d or draw workplane.

![dune3d-no-controls](https://github.com/dune3d/dune3d/assets/3068843/33747704-1376-40f1-823a-1c79d2766b4b)

**Comment by carrotIndustries** [comment, marked as answer]
> I realized that there was a missing step for "Create a new sketch

In https://docs.dune3d.org/en/latest/simple-part.html#create-workplane, there's 

> First, create a new sketch from the plus icon in the workspace browser. [...]

Any suggestions to make this more clear?

> Now that my workplane is aligned, I can't Create Regular Polygon. 

The current sketch has no active workplane. Right click the newly-created workplane and select "Set workplane".

I think it's probably a good idea to automatically set a newly-created workplane as the sketch's active workplane since that's what most users want in the first place.

**Comment by osresearch** [comment]
Ah, it was a reading comprehension failure on my part...  I had missed those steps since they didn't have images (and was mostly following the images).

It might be helpful to call out the errors for other people who end up in the same state. Perhaps something like:

> This tool requires you then click on the the line that corresponds to the workplane’s horizontal direction followed by a second line to define the plane. The workplane’s normal will then be perpendicular to both of the selected lines. If you get an error "`please click a line from a previous group`", this is because the reference lines can not lie in the current sketch and you need to have created a new sketch.

and

> With the new workplane in place, we can proceed with the sketch for the pocket. Start by drawing a hexagon with the “Draw regular polygon” tool.  If the line drawing and polygon controls are grayed out, you need to select `Set Workplane` in the workplane's context menu to make it the active workplane.

**Reply by carrotIndustries** [reply]
Thanks, updated the tutorial.

**Comment by osresearch** [comment]
![Image of adding the workplane](https://docs.dune3d.org/en/latest/_images/constrain-workplane-normal.png)

Coming back to the tutorial (after two years!?), I had this problem again.  What I realized is that I was clicking on the construction line that I had just created (which is in the new sketch), *not* the horizontal (from the earlier workplane and hightlighted in brown above) and the vertical (from the extrusion).   It might be helpful to have both of those highlighted in the image to show where to click and in which order.
